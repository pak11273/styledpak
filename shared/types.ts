import {
  AlignItemsProps,
  AlignSelfProps,
  BorderRadiusProps,
  ButtonStyleProps,
  ColorProps,
  DisplayProps,
  FlexboxProps,
  FontSizeProps,
  GridProps,
  HeightProps,
  LayoutProps,
  LineHeightProps,
  MaxHeightProps,
  MaxWidthProps,
  MinHeightProps,
  MinWidthProps,
  OrderProps,
  OverflowProps,
  SpaceProps,
  TypographyProps,
  WidthProps,
} from "styled-system"

export type SystemProps = AlignSelfProps &
  AlignItemsProps &
  AlignSelfProps &
  ButtonStyleProps &
  BorderRadiusProps &
  ColorProps &
  DisplayProps &
  FlexboxProps &
  FontSizeProps &
  GridProps &
  HeightProps &
  LayoutProps &
  LineHeightProps &
  MaxHeightProps &
  MaxWidthProps &
  MinHeightProps &
  MinWidthProps &
  OrderProps &
  OverflowProps &
  SpaceProps &
  TypographyProps &
  WidthProps
