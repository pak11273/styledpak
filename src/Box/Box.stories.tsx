import Box from "./Box"
import React from "react"
import { storiesOf } from "@storybook/react"
import { withInfo } from "@storybook/addon-info"
import { withKnobs } from "@storybook/addon-knobs"

export default {
  title: "Box",
}

export const Base = () => (
  <Box p={3} bg="whites.10" color="orange">
    This is a box
  </Box>
)

storiesOf("Box", module).addDecorator(withKnobs)
