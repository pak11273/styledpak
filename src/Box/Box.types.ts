export interface BoxProps {
    foo: string;
}
