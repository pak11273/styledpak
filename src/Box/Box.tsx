import { color, fontSize, layout, space, width } from "styled-system"

import { BoxProps } from "./Box.types"
import React from "react"
import styled from "styled-components"

const Box = styled.div`
  background: ${({ theme }) => theme.colors.primary};
  /* ${color}
  ${fontSize}
  ${space}
  ${width} */
`
export default Box
