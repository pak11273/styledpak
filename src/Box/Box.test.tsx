import Box from "./Box"
import { BoxProps } from "./Box.types"
import React from "react"
import { render } from "@testing-library/react"

describe("Test Component", () => {
  let props: BoxProps

  beforeEach(() => {
    props = {
      foo: "bar",
    }
  })

  const renderComponent = () => render(<Box {...props} />)

  it("should render a box", () => {
    expect(renderComponent).toMatchSnapshot()
  })
})
