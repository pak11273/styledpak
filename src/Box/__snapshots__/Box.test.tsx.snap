// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component should render a box 1`] = `[Function]`;
